build.assets/tooling/cmd/update-plist-version/main.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Command update-plist-version updates the version fields of a bundle plist.
+//
+// The version fields updated in an Info.plist file are the CFBundleVersion and
+// CFBundleShortVersionString fields. If the version is not valid as per the
+// Apple specification then the fields will be set to 1.0. A valid version is 3
+// positive integers separated by dots. A semver with a pre-release tag is not
+// valid.
+//
+// This is intended to be used on the tsh.app Info.plist files. Standard
+// releases have the required version number. Pre-releases do not.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"howett.net/plist"
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Fprintf(os.Stderr, "Usage: %s <version> <plist-file>...\n", filepath.Base(os.Args[0]))
+		os.Exit(1)
+	}
+
+	// A version can only be three positive integers separated by periods.
+	versionRE := regexp.MustCompile(`^[0-9]+\.[0-9]+\.[0-9]+$`)
+	version := os.Args[1]
+	if !versionRE.MatchString(version) {
+		version = "1.0"
+	}
+
+	for _, filename := range os.Args[2:] {
+		err := replaceFile(filename, func(in, out *os.File) error {
+			return updateVersion(version, in, out)
+		})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "could not update version of %s: %v\n", filename, err)
+			os.Exit(1)
+		}
+	}
+}
+
+func updateVersion(version string, in, out *os.File) error {
```

Comment:
```suggestion
func updateVersion(version string, in io.Reader, out io.Writer) error {
```

We are not unit testing, but that would make it much simpler to test.

Reply:
I did this originally, but it doesn't work because the plist decoder needs an `io.ReadSeeker`. So I could make `in` be an `io.ReadSeeker` which would look odd and not make sense for the more general `replaceFile`, I could type assert the `io.Reader` into an `io.ReadSeeker` in the closure which makes an assumption that may break the non-existent tests, or just go with `*os.File` - I chose that, and I think I still prefer it. It could be addressed if someone felt like writing tests for this tiny program.

Reply:
I have reconsidered this and followed your suggestion. I've used a type assertion with a guarantee that the type assertion will not panic.